chia/pools/pool_wallet.py

```diff
@@ -784,15 +821,15 @@ async def claim_pool_rewards(self, fee: uint64) -> TransactionRecord:
             fee_amount=fee,
             confirmed=False,
             sent=uint32(0),
-            spend_bundle=spend_bundle,
-            additions=spend_bundle.additions(),
-            removals=spend_bundle.removals(),
+            spend_bundle=full_spend,
+            additions=full_spend.additions(),
+            removals=full_spend.removals(),
             wallet_id=uint32(self.wallet_id),
             sent_to=[],
             memos=[],
             trade_id=None,
             type=uint32(TransactionType.OUTGOING_TX.value),
-            name=spend_bundle.name(),
+            name=full_spend.name(),
         )
         await self.wallet_state_manager.add_pending_transaction(absorb_transaction)
```

Comment:
You should be pushing both transaction records otherwise the balance will just disappear from the wallet without a trace.  You can use `dataclasses.replace` to set the spend bundle of the fee tx to `None` so that when you push it, it doesn't push a duplicate spend.

Also, out of the scope of this PR but `generate_signed_transaction` in the other wallets doesn't push transactions for you.  It only generates the transaction and then the RPC pushes it or whatever.  What this should probably return is a `List[TransactionRecord]` containing both the pool tx and the fee tx and leave it to the calling function to push them for you.  But that's for another day.

Reply:
Interesting. So you're saying that `self.wallet_state_manager.add_pending_transaction` doesn't take into consideration that the `spend_bundle` parameter contains spends concerning two different wallets, and I should make two TransactionRecords, one for `fee_spend`, and one for the claim spend?

But if I do that, doesn't the PoolWallet need the tx id of the combinded transaction, full_spend?

Ahhh, no, you are saying that I should submit a TransactionRecord for the fee_spend only, for the main wallet, and a combined tx record for the full transaction, but without the fee, for the PoolWallet, so the balances are computed correctly. Is that correct?

Reply:
Right now you have this `fee_spend` variable above and you just take its spend bundle and aggregate it with the absorb spend to create a new `full_spend` which you attach to the absorb transaction record.  This is all well and good, but you want to push the transaction record that the standard wallet generated for the fee so it correctly shows up in that wallet's history.  The problem is that you've already aggregated the spend bundle from the fee tx into the absorb tx (because you had to) and so if you try to push the fee tx it will fail since it conflicts with `full_spend` which has already been pushed.  For this reason, you need to set `spend_bundle=None` on the fee tx before you push it.

Reply:
@Quexington  Can you take a look at the current updated code?